src/stack/Stack.java

```diff
@@ -19,38 +19,49 @@ public Stack(int size) {
 	 * Add element into stack
 	 */
 	public void push(int value) {
-
+		if (tos == data.length - 1) {
+			ensureCapacity();
+		}
+		data[++tos] = value;
 	}
 
 	/**
 	 * Get element from the stack
 	 * @return
 	 */
 	public int pop() {
-
-		return 0;
+		if (isEmpty()) {
+			return -1;
+		} else {
+			return data[tos--];
+		}
 	}
 
 
 	/**
 	 * Clear stack
 	 */
 	public void clear() {
-
+		tos = -1;
 	}
 
 	/**
 	 * Check if stack is empty
 	 * @return
 	 */
 	public boolean isEmpty() {
+		if (tos < 0) {
```

Comment:
Your code is correct, but you can just put **return tos == -1;** inside the method.